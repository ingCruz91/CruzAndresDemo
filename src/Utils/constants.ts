export const PRODUCT_LABELS = {
  BIENVENIDO_DE_VUELTA: 'Bienvenido de Vuelta!!',
  NOMBRE_USUARIO: 'Ruben Rodriguez',
  TUS_PUNTOS: 'TUS PUNTOS',
  MES: 'Diciembre',
  MOVIMIENTOS: 'MOVIMIENTOS',
  GANADOS: 'Ganados',
  CANJEADOS: 'Canjeados',
  TODOS: 'Todos',
  DETALLES_DEL_PRODUCTO: 'Detalles del Producto',
  CON_ESTA_COMPRA_ACUMULASTE: 'Con esta compra acumulaste:',
  ACEPTAR: 'Aceptar',
  PTS: '  pts',
  PUNTOS: ' puntos',
};

export const PRODUCT_API_URL =
  'https://6222994f666291106a29f999.mockapi.io/api/v1/products';
