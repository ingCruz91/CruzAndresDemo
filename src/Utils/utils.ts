function formatDate(dateString: string): string {
  const options: Intl.DateTimeFormatOptions = {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  };
  const date: Date = new Date(dateString);
  return date.toLocaleDateString('es-ES', options);
}

export default formatDate;
