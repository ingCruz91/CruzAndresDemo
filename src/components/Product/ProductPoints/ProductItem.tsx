import {StyleSheet, TouchableOpacity, View} from 'react-native';
import CustomLabel from '../../custom/CustomLabel';
import React from 'react';
import formatDate from '../../../Utils/utils';
import {Product} from '../../../screens/ProductPoints';
import {useNavigation} from '@react-navigation/native';
import {StackNavigationProp} from '@react-navigation/stack';
import {RootStackParamList} from '../../../navigation/Navigation';
import FastImage from 'react-native-fast-image';

function ProductItem(itemData: Product) {
  const navigation = useNavigation<StackNavigationProp<RootStackParamList>>();

  const {image, product, createdAt, is_redemption, points} = itemData;

  const formatDateCreatedAt = formatDate(createdAt);

  function navigateToProductPointsDetails() {
    navigation.navigate('ProductPointsDetails', {
      productName: product,
      productCreatedAt: formatDateCreatedAt,
      productPoints: points,
      productImage: image,
    });
  }

  return (
    <View>
      <TouchableOpacity
        testID="productPress"
        onPress={navigateToProductPointsDetails}
        style={styles.container}>
        <FastImage
          testID="productImage"
          style={styles.image}
          source={{
            uri: image,
            priority: FastImage.priority.normal,
          }}
          resizeMode={FastImage.resizeMode.contain}
        />
        <View style={styles.productContainer}>
          <CustomLabel.BlackBold size={14} style={styles.product}>
            {product}
          </CustomLabel.BlackBold>
          <CustomLabel.BlackRegular size={12} style={styles.createdAt}>
            {formatDateCreatedAt}
          </CustomLabel.BlackRegular>
        </View>
        <View style={styles.pointsContainer}>
          <CustomLabel
            style={
              is_redemption ? styles.redemptionTrue : styles.redemptionFalse
            }>
            {is_redemption ? '+' : '-'}
          </CustomLabel>
          <CustomLabel.BlackBold style={styles.points}>
            {points}
          </CustomLabel.BlackBold>
          <CustomLabel.BlackBold style={styles.icon}>
            {'>'}
          </CustomLabel.BlackBold>
        </View>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginTop: 10,
    paddingHorizontal: 10,
  },
  image: {
    borderRadius: 5,
    width: 50,
    height: 50,
    marginRight: 10,
  },
  productContainer: {
    marginTop: 5,
    flex: 1,
  },
  pointsContainer: {
    flexDirection: 'row',
  },
  createdAt: {
    marginTop: 4,
  },
  redemptionTrue: {
    fontSize: 14,
    fontWeight: 'bold',
    alignSelf: 'center',
    color: 'green',
  },
  redemptionFalse: {
    fontSize: 14,
    fontWeight: 'bold',
    alignSelf: 'center',
    color: 'red',
  },
  product: {
    color: '#000000',
  },
  points: {
    fontSize: 14,
    color: 'black',
    fontWeight: 'bold',
    alignSelf: 'center',
  },
  icon: {
    fontSize: 14,
    color: 'black',
    fontWeight: 'bold',
    marginLeft: 15,
    alignSelf: 'center',
  },
});

export default ProductItem;
