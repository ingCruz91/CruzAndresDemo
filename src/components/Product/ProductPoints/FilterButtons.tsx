import React from 'react';
import {StyleSheet, TouchableOpacityProps, View} from 'react-native';
import CustomButton from '../../custom/CustomButton';
import {PRODUCT_LABELS} from '../../../Utils/constants';

interface FilterButtons extends TouchableOpacityProps {
  onPressEarned: () => void;
  onPressRedeemed: () => void;
}

const FilterButtons = ({onPressEarned, onPressRedeemed}: FilterButtons) => {
  return (
    <View style={styles.buttonsContainer}>
      <CustomButton.Primary
        style={styles.buttonStyle}
        onPress={() => onPressEarned()}>
        {PRODUCT_LABELS.GANADOS}
      </CustomButton.Primary>
      <CustomButton.Primary
        style={styles.buttonStyle}
        onPress={onPressRedeemed}>
        {PRODUCT_LABELS.CANJEADOS}
      </CustomButton.Primary>
    </View>
  );
};

const styles = StyleSheet.create({
  buttonsContainer: {
    flexDirection: 'row',
    flex: 1,
    alignItems: 'center',
  },
  buttonStyle: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
    height: 45,
    flex: 1,
    marginHorizontal: 15,
  },
});

export default FilterButtons;
