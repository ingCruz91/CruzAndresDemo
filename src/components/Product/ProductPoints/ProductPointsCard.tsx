import React from 'react';
import {StyleSheet, View} from 'react-native';
import CustomLabel from '../../custom/CustomLabel';
import {PRODUCT_LABELS} from '../../../Utils/constants';

interface ProductPoints {
  totalPoints: string | undefined;
}
const ProductPointsCard = ({totalPoints}: ProductPoints) => {
  return (
    <View style={styles.pointsContainer}>
      <CustomLabel.GrayBold size={14} style={styles.yourPointsLabel}>
        {PRODUCT_LABELS.TUS_PUNTOS}
      </CustomLabel.GrayBold>
      <View style={styles.pointsCard}>
        <CustomLabel.WhiteBold size={16} style={styles.yourPointsText}>
          {PRODUCT_LABELS.MES}
        </CustomLabel.WhiteBold>
        <CustomLabel.WhiteBold size={29}>
          {totalPoints + PRODUCT_LABELS.PTS}
        </CustomLabel.WhiteBold>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  pointsContainer: {
    flex: 2.4,
    alignItems: 'stretch',
  },
  pointsCard: {
    shadowOpacity: 0.5,
    shadowRadius: 5,
    shadowOffset: {
      width: 0,
      height: 6,
    },
    elevation: 5,
    backgroundColor: '#334FFA',
    alignItems: 'center',
    marginHorizontal: 50,
    flex: 1,
    marginBottom: 10,
    borderRadius: 20,
  },
  yourPointsLabel: {
    alignSelf: 'flex-start',
    fontSize: 14,
    marginLeft: 15,
    marginTop: 10,
    marginBottom: 15,
  },
  yourPointsText: {
    alignSelf: 'flex-start',
    marginTop: 15,
    marginLeft: 19,
    marginBottom: 10,
  },
});

export default ProductPointsCard;
