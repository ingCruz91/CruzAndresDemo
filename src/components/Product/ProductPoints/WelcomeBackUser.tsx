import React from 'react';
import {StyleSheet, View} from 'react-native';
import CustomLabel from '../../custom/CustomLabel';
import {PRODUCT_LABELS} from '../../../Utils/constants';

const WelcomeBackUser = () => {
  return (
    <View style={styles.welcomeContainer}>
      <CustomLabel.BlackBold size={16}>
        {PRODUCT_LABELS.BIENVENIDO_DE_VUELTA}
      </CustomLabel.BlackBold>
      <CustomLabel.BlackRegular>
        {PRODUCT_LABELS.NOMBRE_USUARIO}
      </CustomLabel.BlackRegular>
    </View>
  );
};

const styles = StyleSheet.create({
  welcomeContainer: {
    flex: 0.6,
    marginLeft: 15,
  },
});

export default WelcomeBackUser;
