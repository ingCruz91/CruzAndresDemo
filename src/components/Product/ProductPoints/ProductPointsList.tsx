import React from 'react';
import {FlatList, StyleSheet, View} from 'react-native';
import CustomLabel from '../../custom/CustomLabel';
import ProductItem from './ProductItem';
import {Product} from '../../../screens/ProductPoints';
import {PRODUCT_LABELS} from '../../../Utils/constants';

interface ProductPointsList {
  data: Product[] | null;
}
const ProductPointsList = ({data}: ProductPointsList) => {
  return (
    <View style={styles.userPointsContainer}>
      <CustomLabel.GrayBold size={14} style={styles.userPointsSection}>
        {PRODUCT_LABELS.MOVIMIENTOS}
      </CustomLabel.GrayBold>
      <View style={styles.userPointsList}>
        <FlatList
          testID="flatListTest"
          data={data}
          renderItem={itemData => {
            return <ProductItem {...itemData.item} />;
          }}
          initialNumToRender={5}
          maxToRenderPerBatch={5}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  userPointsContainer: {
    borderRadius: 10,
    flex: 4.5,
  },
  userPointsSection: {
    alignSelf: 'flex-start',
    fontSize: 14,
    marginLeft: 15,
    marginTop: 10,
    marginBottom: 15,
  },
  userPointsList: {
    flex: 1,
    backgroundColor: 'white',
    marginBottom: 10,
    borderRadius: 20,
    paddingTop: 20,
    paddingBottom: 20,
    marginHorizontal: 15,
  },
});

export default ProductPointsList;
