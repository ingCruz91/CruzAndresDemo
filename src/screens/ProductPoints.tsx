import React, {useState} from 'react';
import {StyleSheet, SafeAreaView} from 'react-native';
import ProductPointsCard from '../components/Product/ProductPoints/ProductPointsCard';
import ProductPointsList from '../components/Product/ProductPoints/ProductPointsList';
import FilterButtons from '../components/Product/ProductPoints/FilterButtons';
import useProductData from '../hooks/useProductData';
import ProductButton from '../components/Product/ProductButton';
import WelcomeBackUser from '../components/Product/ProductPoints/WelcomeBackUser';
import {PRODUCT_API_URL, PRODUCT_LABELS} from '../Utils/constants';

export interface Product {
  createdAt: string;
  product: string;
  points: number;
  image: string;
  is_redemption: boolean;
  id: string;
}

function ProductPoints() {
  const productData = useProductData(PRODUCT_API_URL);

  const [filteredData, setFilteredData] = useState<Product[] | null>([]);
  const [enableFilterButtons, setEnableFilterButtons] = useState<boolean>(true);

  const earnedData = productData?.filter(product => product?.is_redemption);
  const redeemedData = productData?.filter(product => !product?.is_redemption);

  function getTotalPoints() {
    const points = redeemedData?.map(
      product => ({points: product?.points} || []),
    );
    const totalPoints = points?.reduce((acc, curr) => acc + curr?.points, 0);

    return (
      totalPoints?.toLocaleString('en-US', {
        style: 'decimal',
      }) || '0'
    );
  }

  function setFilteredEarnedData() {
    setEnableFilterButtons(!enableFilterButtons);
    if (earnedData) {
      setFilteredData(earnedData);
    }
  }

  function setFilteredRedeemedData() {
    setEnableFilterButtons(!enableFilterButtons);
    if (redeemedData) {
      setFilteredData(redeemedData);
    }
  }

  return (
    <SafeAreaView style={styles.appContainer}>
      <WelcomeBackUser />
      <ProductPointsCard totalPoints={getTotalPoints()} />
      <ProductPointsList
        data={enableFilterButtons ? productData : filteredData}
      />
      {enableFilterButtons ? (
        <FilterButtons
          testID="filter-buttons"
          onPressEarned={setFilteredEarnedData}
          onPressRedeemed={setFilteredRedeemedData}
        />
      ) : (
        <ProductButton
          testID="product-button"
          onPress={() => setEnableFilterButtons(!enableFilterButtons)}>
          {PRODUCT_LABELS.TODOS}
        </ProductButton>
      )}
    </SafeAreaView>
  );
}
const styles = StyleSheet.create({
  appContainer: {
    backgroundColor: '#EAEAEA',
    flex: 1,
    borderWidth: 1,
    borderColor: 'pink',
  },
});

export default ProductPoints;
