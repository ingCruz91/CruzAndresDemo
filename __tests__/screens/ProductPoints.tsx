import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import ProductPoints from '../../src/screens/ProductPoints';
import {PRODUCT_LABELS} from '../../src/Utils/constants';

describe('ProductPoints', () => {
  it('renders ProductPoints correctly', () => {
    const {toJSON} = render(<ProductPoints />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should display all points button when earned points button is pressed', async () => {
    const {getByText} = render(<ProductPoints />);

    const earnedButton = getByText(PRODUCT_LABELS.GANADOS);

    fireEvent.press(earnedButton);
    expect(getByText(PRODUCT_LABELS.TODOS)).toBeTruthy();
  });

  it('should display all points button when redeemed points button is pressed', async () => {
    const {getByText} = render(<ProductPoints />);

    const redeemedButton = getByText(PRODUCT_LABELS.CANJEADOS);

    fireEvent.press(redeemedButton);
    expect(getByText(PRODUCT_LABELS.TODOS)).toBeTruthy();
  });

  it('should display filter buttons when all points button is pressed', async () => {
    const {getByText} = render(<ProductPoints />);
    // first need to press any of filter buttons to show all points button
    const earnedButton = getByText(PRODUCT_LABELS.GANADOS);
    fireEvent.press(earnedButton);

    const allPointsButton = getByText(PRODUCT_LABELS.TODOS);
    fireEvent.press(allPointsButton);
    expect(getByText(PRODUCT_LABELS.GANADOS)).toBeTruthy();
  });

  it('renders flatList correctly when component first render', () => {
    const {getByTestId} = render(<ProductPoints />);
    const productPointsList = getByTestId('flatListTest');

    expect(productPointsList).toBeTruthy();
  });

  it('renders flatList correctly when press earned points button', () => {
    const {getByTestId, getByText} = render(<ProductPoints />);
    const productPointsList = getByTestId('flatListTest');

    const earnedButton = getByText(PRODUCT_LABELS.GANADOS);
    fireEvent.press(earnedButton);

    expect(productPointsList).toBeTruthy();
  });

  it('renders flatList correctly when press redeemed points button', () => {
    const {getByTestId, getByText} = render(<ProductPoints />);
    const productPointsList = getByTestId('flatListTest');

    const redeemedButton = getByText(PRODUCT_LABELS.CANJEADOS);
    fireEvent.press(redeemedButton);

    expect(productPointsList).toBeTruthy();
  });

  it('renders flatList correctly when press allPoints Button', () => {
    const {getByTestId, getByText} = render(<ProductPoints />);

    // first need to press any of filter buttons to show all points button
    const earnedButton = getByText(PRODUCT_LABELS.GANADOS);
    fireEvent.press(earnedButton);

    const allPointsButton = getByText(PRODUCT_LABELS.TODOS);
    fireEvent.press(allPointsButton);

    const productPointsList = getByTestId('flatListTest');

    expect(productPointsList).toBeTruthy();
  });
});
