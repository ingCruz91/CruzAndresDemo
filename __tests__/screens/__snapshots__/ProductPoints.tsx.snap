// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPoints renders ProductPoints correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#EAEAEA",
      "borderColor": "pink",
      "borderWidth": 1,
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 0.6,
        "marginLeft": 15,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#020202",
            "fontFamily": "AvenirBlack",
            "fontWeight": "bold",
          },
          {
            "fontFamily": undefined,
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Bienvenido de Vuelta!!
    </Text>
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#020202",
            "fontFamily": "AvenirRoman",
          },
          {
            "fontFamily": undefined,
            "fontSize": undefined,
          },
          undefined,
        ]
      }
    >
      Ruben Rodriguez
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "stretch",
        "flex": 2.4,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#9B9898",
            "fontFamily": "AvenirBlack",
            "fontWeight": "bold",
          },
          {
            "fontFamily": undefined,
            "fontSize": 14,
          },
          {
            "alignSelf": "flex-start",
            "fontSize": 14,
            "marginBottom": 15,
            "marginLeft": 15,
            "marginTop": 10,
          },
        ]
      }
    >
      TUS PUNTOS
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#334FFA",
          "borderRadius": 20,
          "elevation": 5,
          "flex": 1,
          "marginBottom": 10,
          "marginHorizontal": 50,
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 5,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "white",
              "fontFamily": "AvenirBlack",
              "fontWeight": "bold",
            },
            {
              "fontFamily": undefined,
              "fontSize": 16,
            },
            {
              "alignSelf": "flex-start",
              "marginBottom": 10,
              "marginLeft": 19,
              "marginTop": 15,
            },
          ]
        }
      >
        Diciembre
      </Text>
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "white",
              "fontFamily": "AvenirBlack",
              "fontWeight": "bold",
            },
            {
              "fontFamily": undefined,
              "fontSize": 29,
            },
            undefined,
          ]
        }
      >
        0  pts
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "borderRadius": 10,
        "flex": 4.5,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#9B9898",
            "fontFamily": "AvenirBlack",
            "fontWeight": "bold",
          },
          {
            "fontFamily": undefined,
            "fontSize": 14,
          },
          {
            "alignSelf": "flex-start",
            "fontSize": 14,
            "marginBottom": 15,
            "marginLeft": 15,
            "marginTop": 10,
          },
        ]
      }
    >
      MOVIMIENTOS
    </Text>
    <View
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 20,
          "flex": 1,
          "marginBottom": 10,
          "marginHorizontal": 15,
          "paddingBottom": 20,
          "paddingTop": 20,
        }
      }
    >
      <RCTScrollView
        data={null}
        getItem={[Function]}
        getItemCount={[Function]}
        initialNumToRender={5}
        keyExtractor={[Function]}
        maxToRenderPerBatch={5}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={[]}
        testID="flatListTest"
        viewabilityConfigCallbackPairs={[]}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#334FFA",
          "borderRadius": 10,
          "flex": 1,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 15,
          "opacity": 1,
        }
      }
    >
      <Text>
        <Text
          style={
            [
              {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
            ]
          }
        >
          Ganados
        </Text>
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#334FFA",
          "borderRadius": 10,
          "flex": 1,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 15,
          "opacity": 1,
        }
      }
    >
      <Text>
        <Text
          style={
            [
              {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
            ]
          }
        >
          Canjeados
        </Text>
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
